### PR TITLE
use ruff as linter and fix lots of problems

### PR DESCRIPTION
**Pull request by FHU-yezi**
### All Submissions:

- [x] Have you followed the guidelines stated in [CONTRIBUTING.md](https://github.com/pynecone-io/pynecone/blob/main/CONTRIBUTING.md) file?
- [x] Have you checked to ensure there aren't any other open [Pull Requests](https://github.com/pynecone-io/pynecone/pulls ) for the desired changed?

### Type of change

Please delete options that are not relevant.

- [x] This change need a CI pipeline update. ( I have already changed the build workflow )
- [x] This change is about a development tools change.

### New Feature Submission:

- [x] Does your submission pass the tests? 
- [x] Have you linted your code locally prior to submission?

### Changes To Core Features:

- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your core changes, as applicable?
- [x] Have you successfully ran tests with your changes locally?

### **After** these steps, you're ready to open a pull request.

    a. Give a descriptive title to your PR.

    b. Describe your changes.

    c. Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

[Ruff](https://github.com/charliermarsh/ruff) is a super-fast Python code linter written in Rust. This Pull Request removed the old pydocstyle linter and isort import reorder tool, the ruff documentation said it can be a replacement of these tools.

I also changed `.github/workflows/build.yml` file to let the CI pipeline running correctly ( maybe? ), and `CONTRIBUTING.md` to replace pydocstyle lines with ruff check things.

With the help from ruff, I changed lots of buggy or unstable code in our code base, included some unused import in non-init files, some meaningless f-string and some missing chars in our docstring.

For editor integration, ruff has a official VS Code Extension and an unofficial PyCharm extension, Vim guys can also use official LSP called ruff-lsp.

For a performance reference, I use hyperfine to benchmark ruff ( with all the features we need turned on ) and flake8 ( with zero plugin ):

```bash
❯ hyperfine "ruff ." "flake8 ." -i
Benchmark 1: ruff .
  Time (mean ± σ):      10.6 ms ±   5.1 ms    [User: 8.4 ms, System: 5.9 ms]
  Range (min … max):     7.7 ms …  68.3 ms    202 runs
 
  Warning: Statistical outliers were detected. Consider re-running this benchmark on a quiet PC without any interferences from other programs. It might help to use the '--warmup' or '--prepare' options.
 
Benchmark 2: flake8 .
  Time (mean ± σ):      1.253 s ±  0.011 s    [User: 3.515 s, System: 0.110 s]
  Range (min … max):    1.241 s …  1.279 s    10 runs
 
  Warning: Ignoring non-zero exit code.
 
Summary
  'ruff .' ran
  118.42 ± 57.40 times faster than 'flake8 .'
```

Speed up development and CI, make more features with less bugs, let's "rust" it up~

### CODE REVIEW

**Comment by Alek99** [comment]
Is this the same as black?

**Comment by Alek99** [comment]
I like this addition. At first glance it looks good I'm going to pull this branch and test it locally tomorrow and then ill approve

**Comment by Alek99** [comment]
I think there are some merge conflicts can you resolve these and then I'll merge approve looks good

**Comment by FHU-yezi** [comment]
@Alek99 Have merged all the change from upstream main branch, please review again and we are about to go.

**Comment by Alek99** [comment]
Ok I just tried running this code and got an error,

```
./pages/index.js
Error: 
  x Unexpected token `{`. Expected identifier, string literal, numeric literal or [ for the computed key
    ,----
 35 | {set_state}({{
    :              ^
    `----

Caused by:
    0: failed to process input file
    1: Syntax Error
```

I think when ruff took out the f strings its causing an error. Does this work locally for you

**Comment by FHU-yezi** [comment]
I just ran the unit test and it passed. Will try to revent this change.

File: pynecone/compiler/templates.py [hunk 1]
```diff
@@ -164,12 +164,18 @@ def format_state(
         "  if(!isReady) {{",
         "    return;",
         "  }}",
-        f"  if (!{SOCKET}.current) {{{{",
-        f"    connect({SOCKET}, {{state}}, {{set_state}}, {RESULT}, {SET_RESULT}, {ROUTER}, {EVENT_ENDPOINT})",
+        "  const reconnectSocket = () => {{",
```
Comment (picklelo) [inline pynecone/compiler/templates.py:167]: I don't think we need this change - seems to be modifying the logic

**Comment by FHU-yezi** [comment]
Fixed this error and ready to review again.

**Review by picklelo — CHANGES_REQUESTED** [review]
This looks great except for pynecone/compiler/templates.py - let's undo all the changes there. The apps won't run with the code as is

**Comment by FHU-yezi** [comment]
@picklelo I use git blame to locate the editor of these lines, seems it is modified by advo-kat in [0056d94d](https://github.com/pynecone-io/pynecone/commit/0056d94d57a52b3fe5a38b60e6734185c79f2115) two weeks ago.

I just try to override this file with latest version in our master branch, please review again, sorry for this problem.

**Review by picklelo — APPROVED** [review]
Awesome thanks for fixing!

**Pull request merged**